app/src/main/java/org/fossasia/openevent/app/utils/CountryUtils.java

```diff
@@ -57,33 +58,47 @@ private CurrencyUtils() {
             .subscribeOn(Schedulers.computation());
     }
 
-    @SuppressWarnings({ "PMD.AvoidCatchingGenericException", "PMD.EmptyCatchBlock" })
-    public static List<String> getCurrencyCodesList() {
-        List<String> currencyCodes = new ArrayList<>();
-        List<String> priorCurrencyCodes = new ArrayList<>();
-
-        priorCurrencyCodes.add(Currency.getInstance(Locale.US).toString());
-        priorCurrencyCodes.add(Currency.getInstance(Locale.UK).toString());
-        priorCurrencyCodes.add(Currency.getInstance(Locale.GERMANY).toString());
-        priorCurrencyCodes.add(Currency.getInstance(Locale.JAPAN).toString());
-        priorCurrencyCodes.add(Currency.getInstance(Locale.CANADA).toString());
-        priorCurrencyCodes.add(Currency.getInstance(Locale.CHINA).toString());
-        priorCurrencyCodes.add(Currency.getInstance(Locale.KOREA).toString());
-
-        for (Locale locale : Locale.getAvailableLocales()) {
+    public static Map<String, String> getCountryCurrencyMap() {
+        Locale[] locales = Locale.getAvailableLocales();
+        Map<String, String> countryCurrencyMap = new TreeMap<>();
+        String country;
+        Currency currency;
```

Comment:
Why are county and currency defined outside the loop

Reply:
Only the reference is outside the loop. You are not reusing variables to be declared outside. New allocation is still done in the loop

```Java
SharedObject so = new SharedObject();
for (Item i : items {
    so.doSomething();
    // stuff
}
```

This is better

```Java
SharedObject so;
for (Item i : items {
    so = new SharedObject()
    so.doSomething();
    // stuff
}
```
This is not

Also, when there is no object that is loop invariant which is the case here, then

```Java
String name;
String title;
for (Item i : items) {
    name = item.getName();
    title = item.getTitle();
}
```

is worse than

```Java
for (Item i : items) {
    String name = item.getName();
    String title = item.getTitle();
}
```

because it
- has no performance benefit
- increases the scope of variables outside the loop
- makes the variables nullable

Also, it should be noted that simplicity, readability and maintainability are favoured far more than performance. Optimisation is the last thing to be considered after profiling and checking of the bottleneck is done. Adding to this, however, it is important to follow good practices related to performance as you mentioned, but only knowing when they are applied

Reply:
Thanks for sharing your knowledge and for the very detailed answer. I sure am learning a lot from you. I'll make the change and push.